tasks/static_quality_gates/lib/gates_lib.py

```diff
@@ -262,3 +273,350 @@ def generate_metric_reports(self, ctx, filename="static_gate_report.json", branc
                 f"aws s3 cp --only-show-errors --region us-east-1 --sse AES256 {filename} {self.S3_REPORT_PATH}/{CI_COMMIT_SHA}/{filename}",
                 hide="stdout",
             )
+
+
+class StaticQualityGateFailed(Exception):
+    """
+    Exception raised when a static quality gate fails
+    """
+
+    def __init__(self, message: str):
+        self.message = color_message(message, "red")
+        super().__init__(self.message)
+
+
+class StaticQualityGate:
+    """
+    Base class for all static quality gates
+    that contains the common attributes for all static quality gates
+    """
+
+    gate_name: str
+    arch: str
+    os: str
```

Comment:
Since this is going to be computed by subclasses, how about turning into a [(cached) property](https://docs.python.org/3/library/functools.html#functools.cached_property)? The base class could present it as an abstract method / property. Same for `artifact_path`.